### PR TITLE
plumb customer ca

### PR DESCRIPTION
**Pull request by rogeruiz**
This patch series plumbs the work done in hashicorp/ptfe-proxy#2 and hashicorp/ptfe-replicated#259 into the GCP Terraform module.

### CODE REVIEW

File: files/install-ptfe.sh [hunk 1]
```diff
@@ -185,6 +197,15 @@ if [[ $(< /etc/ptfe/airgap-installer-url) != none ]]; then
     airgap_installer_url_path="/etc/ptfe/airgap-installer-url"
 fi
 
+if [[ $(< /etc/ptfe/custom-ca-cert-url) != none ]]; then
```
[the remark this answers is not in the document]
Comment (erindatkinson) [reply files/install-ptfe.sh]: does it make more sense to have this be an empty string default over a "none" string as the default?  to @evanphx 's point about metadata errors, how do we want to handle an empty string failure (eg. api blip during metadata retrieval)

File: files/install-ptfe.sh [hunk 2]
```diff
@@ -185,6 +197,20 @@ if [[ $(< /etc/ptfe/airgap-installer-url) != none ]]; then
...
+  custom_ca_cert_url=$(cat /etc/ptfe/custom-ca-cert-url)
+  custom_ca_cert_file_name=$(echo "${custom_ca_cert_url}" | awk -F '/' '{ print $NF }')
+  # Setting up a tmp directory to do this `jq` transform to leave artifcats if anything goes "boom".
+  mkdir -p /tmp/ptfe/customer-certs
+  pushd /tmp/ptfe/customer-certs
+  wget --trust-server-files "${custom_ca_cert_url}"
```
Comment (bnferguson) [inline files/install-ptfe.sh]: Wondering if we check that the return code here is good and we got something? Perhaps bail with an error message if the wget failed for any reason?

File: files/install-ptfe.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (rogeruiz) [reply files/install-ptfe.sh]: I'm not sure if terraform will allow an `""` value if the default is set. But i'm not sure. @erindatkinson

**Comment by rogeruiz** [comment]
I'm now checking if there's a non-zero length string and that it's equal to something bersides the default string. We should probably standardize that across the other conditional checks, but I won't be doing that in this PR.

**Pull request merged**